### PR TITLE
Agg watcher reduce usage pr

### PR DESCRIPTION
**Pull request by Sindica**
Perf run in https://github.com/futurewei-cloud/arktos-perftest/tree/perf-20201019
CI passed in https://github.com/centaurus-cloud/arktos/pull/784
Compacted commits for easiler review

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/watch/watch.go [hunk 1]
```diff
@@ -20,12 +20,10 @@ package watch
 import (
 	"context"
 	"fmt"
-	"github.com/grafov/bcast"
```
Comment (yb01) [inline staging/src/k8s.io/apimachinery/pkg/watch/watch.go:23]: those are in PR 782 already which is merged. can rebase to remove them.

Comment (Sindica) [reply staging/src/k8s.io/apimachinery/pkg/watch/watch.go:23]: Yes, I based this branch from previous commit before it was merged into master and did not rebase to save CI rerun.

File: staging/src/k8s.io/client-go/tools/watch/retrywatcher.go [hunk 2]
```diff
@@ -101,26 +101,12 @@ func (rw *RetryWatcher) send(event watch.Event) bool {
...
-	}
-
-	defer aggWatchers.Stop()
-	if aggWatchers.GetWatchersCount() == 0 {
-		klog.Error("No watcher in aggregated watchers")
-		return false, 0
-	}
-
-	err := aggWatchers.GetErrors()
 	switch err {
```
Comment (yb01) [inline staging/src/k8s.io/client-go/tools/watch/retrywatcher.go:110]: are the errors from the watcher and aggWatcher.GetErrors() match ?

**Review by yb01 — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/client-go/tools/watch/retrywatcher.go (hunk 2 above)
Comment (Sindica) [reply staging/src/k8s.io/client-go/tools/watch/retrywatcher.go:110]: This is a revert back to previous implementation. Watch returns watch result and error if any. When I changed the return type of Watch interface, I have to aggregate the watch results and errors. Now the implementation of Watch function will decide how to return error. The auto generated client code calls GetErrors to return error.

**Pull request merged**